src/pybind/mgr/orchestrator/_interface.py

```diff
@@ -1292,20 +1295,22 @@ def matches_service(self, service_name):
         return False
 
     def service_id(self):
-        if self.daemon_type == 'rgw':
-            if self.hostname and self.hostname in self.daemon_id:
-                pre, post_ = self.daemon_id.split(self.hostname)
-                return pre[:-1]
-            else:
-                # daemon_id == "realm.zone.host.random"
-                v = self.daemon_id.split('.')
-                if len(v) == 4:
-                    return '.'.join(v[0:2])
-                # subcluster or fqdn? undecidable.
-                raise OrchestratorError(f"DaemonDescription: Cannot calculate service_id: {v}")
```

Comment:
can we keep this special handing for RGWs?

Reply:
do we need to keep it around for some reason?

Reply:
It's just we end up in an endless loop, if we mess up this method. I'm just a bit nervous.